### PR TITLE
Re worked whole map view system, and added TileMapTool system

### PR DESCRIPTION
**Pull request by Bdtrotte**

### CODE REVIEW

**Review by mobile-bungalow — APPROVED** [review]
*thumb* 

this is a super clean rework. can't believe I forgot the destructor, I also wouldn't have guessed that removeItem free's the data.

File: WallsAndHoles/tile.cpp [hunk 1]
```diff
@@ -61,18 +64,25 @@ void Tile::setRelativeHeight(float relativeHeight)
 
 void Tile::setRelativePosition(QVector2D relavtivePosition)
```
Comment (timoffex) [inline WallsAndHoles/tile.cpp:65]: relativePosition misspelled

File: WallsAndHoles/tilemaptoolmanager.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#include "tilemaptoolmanager.h"
+
+TileMapToolManager::TileMapToolManager(QObject *parent)
+    : ToolManager(parent) {}
+
+void TileMapToolManager::setTileMap(TileMap *tileMap)
+{
+    for (AbstractToolP t : mTools)
+        static_cast<AbstractTileMapTool *>(t.data())->setTileMap(tileMap);
```
Comment (timoffex) [inline WallsAndHoles/tilemaptoolmanager.cpp:9]: I recommend putting these static casts away into a helper method. It will make the code much clearer.

**Review by timoffex — APPROVED** [review]
Everything looks good, the static_cast and relativePosition things will be dealt with later.

**Pull request merged**